lib/Runtime/Types/TypePropertyCache.cpp

```diff
@@ -172,7 +172,9 @@ namespace Js
                 &propertyIndex,
                 &isInlineSlot,
                 &isMissing,
-                &prototypeObjectWithProperty))
+                &prototypeObjectWithProperty)
+            || (prototypeObjectWithProperty != nullptr
+                && prototypeObjectWithProperty != propertyObject))
```

Comment:
@suwc , as we discussed, I think we're probably abusing the cache on a byte code level. I'll make some time to investigate today. If that's the case, then we want to fix the code gen rather than working around it in the cache logic.